opentelemetry/src/test/scala/zio/telemetry/opentelemetry/TracingTest.scala

```diff
@@ -18,13 +18,14 @@ import zio._
 
 import scala.collection.mutable
 import scala.jdk.CollectionConverters._
+import io.opentelemetry.common.Attributes
 
 object TracingTest extends DefaultRunnableSpec {
 
   val inMemoryTracer: UIO[(InMemoryTracing, Tracer)] = for {
     tracerProvider  <- UIO(TracerSdkProvider.builder().build())
     inMemoryTracing <- UIO(InMemoryTracing.builder().setTracerProvider(tracerProvider).build())
-    tracer          = tracerProvider.get("TracingTest")
+    tracer          = tracerProvider.get("TracingTest").asInstanceOf[Tracer]
```

Comment:
Why do we need to cast here?

Reply:
To be honest, I feel this is some weird behavior on OpenTelemetry Java side. Without this cast, a runtime exception occurs on this particular line:
```
java.lang.IllegalAccessError: failed to access class io.opentelemetry.sdk.trace.TracerSdk from class zio.telemetry.opentelemetry.TracingTest$ (io.opentelemetry.sdk.trace.TracerSdk and zio.telemetry.opentelemetry.TracingTest$ are in unnamed module of loader sbt.internal.LayeredClassLoader @35f299f4)
```
which could not be resolved, even by adding an explicit dependency on the `opentelemetry-sdk` package.

Reply:
OK, let's keep the cast. Could you please add a comment on why it it there (basically just copy and pasting your reply?)